### DISCUSSION TITLE
How to keep passwords secure inside the values.yaml in Git?

### DISCUSSION THREAD

**Discussion by EsDmitrii** [goharbor/harbor#19288] [Q&A] [answered]
So I have a question
When I override default values, I set my custom admin pass, I set custom creds for my external S3, I set custom creds for external Redis, etc
How can I keep it secure in a Git repo?
I tried to keep all stuff inside Hashicorp Vault, but some creds are templatized and encoded by helpers.tpl or resource templates:
<img width="797" alt="Screenshot 2023-09-01 at 15 07 01" src="https://github.com/goharbor/harbor/assets/124890665/21cbab78-fba7-40bf-8a9c-9ce9e60e730c">
<img width="1116" alt="Screenshot 2023-09-01 at 15 07 49" src="https://github.com/goharbor/harbor/assets/124890665/b74a8c78-8ec5-457d-adca-4f6b3d9d9e1b">
<img width="1031" alt="Screenshot 2023-09-01 at 15 08 25" src="https://github.com/goharbor/harbor/assets/124890665/18caa96c-7914-4e2c-a6cb-c5fb013bcdde">

So when I try to set something like this (paste the link to the vault), I get error because templater renders all templates using my values, make them encoded, masked, etc. After all manupulations links to the vault are broken and can't be replaced by ArgoCD with correct values.
 
<img width="466" alt="Screenshot 2023-09-01 at 15 11 20" src="https://github.com/goharbor/harbor/assets/124890665/98ccceb1-5b65-42a4-9881-54dc390a0fa2">

I tried to clone chart, remove all base64 encoding, moved secrets from type `Data` to `stringData`
All seems to be well, but I started to get weird issues described here  https://github.com/goharbor/harbor/issues/19245

Any thoughts?

**Comment by stonezdj** [comment, marked as answer]
It is a general issue, basically you could try to create predefined secret outside, and use the existing secret in values.yaml.

**Reply by EsDmitrii** [reply]
got it 
thank you for the reply
